resilience4j-spring/src/main/java/io/github/resilience4j/decorators/configure/DecoratorsAspect.java

```diff
@@ -0,0 +1,165 @@
+ *         .withRetry(retryRegistry.retry(""))
+ *         .get();
+ * </code></pre>
+ */
+@Aspect
+public class DecoratorsAspect implements Ordered {
+
+    private final RetryAspectHelper retryAspectHelper;
+    private final RateLimiterAspectHelper rateLimiterAspectHelper;
+    private final BulkheadAspectHelper bulkheadAspectHelper;
+    private final CircuitBreakerAspectHelper circuitBreakerAspectHelper;
+    private final int order;
+
+    public DecoratorsAspect(
+        RetryAspectHelper retryAspectHelper,
+        RateLimiterAspectHelper rateLimiterAspectHelper,
+        BulkheadAspectHelper bulkheadAspectHelper,
+        CircuitBreakerAspectHelper circuitBreakerAspectHelper,
+        int order) {
+        this.retryAspectHelper = retryAspectHelper;
+        this.rateLimiterAspectHelper = rateLimiterAspectHelper;
+        this.bulkheadAspectHelper = bulkheadAspectHelper;
+        this.circuitBreakerAspectHelper = circuitBreakerAspectHelper;
+        this.order = order;
+    }
+
+    @Pointcut(
+        value = "@within(decorator) || @annotation(decorator)",
+        argNames = "decorator")
+    public void matchAnnotatedClassOrMethod(Decorator decorator) {
+    }
+
+    @Pointcut(
+        value = "@within(decorators) || @annotation(decorators)",
+        argNames = "decorators")
+    public void matchMultipalyAnnotatedClassOrMethod(Decorators decorators) {
+    }
+
+    @Around(
+        value = "matchAnnotatedClassOrMethod(foundDecoratorAnnotation)",
+        argNames = "proceedingJoinPoint, foundDecoratorAnnotation")
+    public Object decorateAroundSingleAnnotationAdvice(
+        ProceedingJoinPoint proceedingJoinPoint,
+        @Nullable Decorator foundDecoratorAnnotation) throws Throwable {
+        return decorateAroundAdvice(proceedingJoinPoint);
+    }   
+
+    @Around(
+        value = "matchMultipalyAnnotatedClassOrMethod(foundDecoratorsAnnotation)",
+        argNames = "proceedingJoinPoint, foundDecoratorsAnnotation")
+    public Object decorateAroundMultipalyAnnotatedAdvice(
+        ProceedingJoinPoint proceedingJoinPoint,
+        @Nullable Decorators foundDecoratorsAnnotation) throws Throwable {
+        return decorateAroundAdvice(proceedingJoinPoint);
+    }
+    
+    private Object decorateAroundAdvice(ProceedingJoinPoint proceedingJoinPoint) throws Throwable {
+        ProceedingJoinPointHelper joinPointHelper
+            = ProceedingJoinPointHelper.prepareFor(proceedingJoinPoint);
+        List<Decorator> decoratorAnnotations = joinPointHelper.getMethodAnnotations(
+            Decorator.class);
+        if (decoratorAnnotations == null) {
+            decoratorAnnotations = joinPointHelper.getClassAnnotations(Decorator.class);
+        }
+        if (decoratorAnnotations == null) { //because annotations wasn't found
+            return proceedingJoinPoint.proceed();
+        }
+        Collections.reverse(decoratorAnnotations);
+        for (Decorator decoratorAnnotation : decoratorAnnotations) {
```

Comment:
Currently the `@Decorator` annotation allows a user to specify a Retry, CircuitBreaker and Bulkhead in one annotation. The `DecoratorsAspect` can handle it, but it should do it in the order as it is now:  `Retry(CircuitBreaker(RateLimiter(Bulkhead(Function))))`

```
Supplier<String> decoratedSupplier = Decorators.ofSupplier(supplier) 
  .withBulkhead(Bulkhead.ofDefaults("name"))     
  .withRateLimiter(RateLimiter.ofDefaults("name"))
  .withCircuitBreaker(CircuitBreaker.ofDefaults("name"))
  .withRetry(Retry.ofDefaults("name"))
```